rpcs3/Emu/Cell/lv2/sys_config.h

```diff
@@ -0,0 +1,433 @@
+	SYS_CONFIG_SERVICE_USER_0x1030   = SYS_CONFIG_SERVICE_USER_BASE + 0x1030,
+};
+
+enum sys_config_service_listener_type : u32 {
+	SYS_CONFIG_SERVICE_LISTENER_ONCE      = 0,
+	SYS_CONFIG_SERVICE_LISTENER_REPEATING = 1
+};
+
+enum sys_config_event_source : u64 {
+	SYS_CONFIG_EVENT_SOURCE_SERVICE = 1,
+	SYS_CONFIG_EVENT_SOURCE_IO      = 2
+};
+
+
+/*
+ * Dynamic-sized struct to describe a sys_config_service_event
+ * We never allocate it - the guest does it for us and provides a pointer
+ */
+struct sys_config_service_event_t {
+	// Handle to the service listener for whom this event is destined
+	be_t<u32> service_listener_handle;
+
+	// 1 if this service is currently registered or unregistered
+	be_t<u32> registered;
+
+	// Service ID that triggered this event
+	be_t<u64> service_id;
+
+	// Custom ID provided by the user, used to uniquely identify service events (provided to sys_config_register_event)
+	// When a service is unregistered, this is the only value available to distinguish which service event was unregistered.
+	be_t<u64> user_id;
+
+	/* if added==0, the structure ends here */
+
+	// Verbosity of this service event (provided to sys_config_register_event)
+	be_t<u64> verbosity;
+
+	// Size of 'data'
+	be_t<u32> data_size;
+
+	// Ignored, seems to be simply 32-bits of padding
+	be_t<u32> padding;
+
+	// Buffer containing event data (copy of the buffer supplied to sys_config_register_service)
+	// NOTE: This buffer size is dynamic, according to 'data_size', and can be 0. Here it is set to 1 since zero-sized buffers are not standards-compliant
+	u8 data[1];
+};
+
+
+/*
+ * Event data structure for SYS_CONFIG_SERVICE_PADMANAGER
+ * This is a guess
+ */
+struct sys_config_padmanager_data_t {
+	be_t<u16> unk[5]; // hid device type ?
+	be_t<u16> vid;
+	be_t<u16> pid;
+	be_t<u16> unk2[6]; // bluetooth address?
+};
+static_assert(sizeof(sys_config_padmanager_data_t) == 26);
+
+
+/*
+ * Global (fxm-managed) sys_config state
+ */
+
+class lv2_config {
+	// LV2 Config mutex
+	shared_mutex m_mutex;
+
+	// Map of LV2 Service Events
+	std::unordered_map<u32, std::weak_ptr<lv2_config_service_event>> events;
+
+public:
+	void initialize() const;
+
+	// Service Events
+	void add_service_event(const std::shared_ptr<lv2_config_service_event>& event);
+	void remove_service_event(u32 id);
+
+	std::shared_ptr<lv2_config_service_event> find_event(u32 id)
+	{
+		reader_lock lock(m_mutex);
+
+		auto it = events.find(id);
+
+		if (it == events.end())
+			return nullptr;
+
+		if (auto event = it->second.lock())
+		{
+			return event;
+		}
+
+		return nullptr;
+	}
+
+	// Utilities
+	static std::shared_ptr<lv2_config> make()
+	{
+		return fxm::make<lv2_config>();
+	}
+
+	static std::shared_ptr<lv2_config> get()
+	{
+		return fxm::get<lv2_config>();
+	}
+};
+
+/*
+ * LV2 Config Handle object, managed by IDM
+ */
+class lv2_config_handle
+{
+public:
+	static const u32 id_base = 0x41000000;
+	static const u32 id_step = 0x100;
+	static const u32 id_count = 2048;
```

Comment:
from where the count is coming from, worth checking if it needs to be allocated from lv2_obj table.
same for other types here.